src/atc/transformers/select_and_cast_columns_transformer_nc.py

```diff
@@ -0,0 +1,41 @@
+from typing import List, Union
+
+import pyspark.sql.functions as F
+import pyspark.sql.types as T
+from pyspark.sql import DataFrame
+
+from atc.etl import TransformerNC
+
+
+class SelectAndCastColumnsTransformerNC(TransformerNC):
+    """
+    This transformer select and cast columns based pyspark schema.
+
+    Attributes:
+    ----------
+        schema : StructType
+            pyspark schema to use for select and cast
+        dataset_input_keys : Union[str, List[str]]
+            list of input dataset keys
+        dataset_output_key : str
+            output dataset key
+    """
+
+    def __init__(
+        self,
+        *,
+        schema: T.StructType,
+        dataset_input_keys: Union[str, List[str]] = None,
+        dataset_output_key: str = None,
+    ):
+        super().__init__(
+            dataset_input_keys=dataset_input_keys,
+            dataset_output_key=dataset_output_key,
+        )
+        self.schema = schema
+
+    def process(self, df: DataFrame) -> DataFrame:
+
+        selectAndCastColumnsList = [F.col(c.name).cast(c.dataType) for c in self.schema]
```

Comment:
Consider using alias: `selectAndCastColumnsList = [F.col(c.name).cast(c.dataType).alias(cname) for c in self.schema]`. I have sometimes experienced issues with casting without aliassing (when upgrading to newer version of spark).